src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -982,48 +982,36 @@ _mongoc_client_recv (mongoc_client_t *client,
 }
 
 
-/*
- *--------------------------------------------------------------------------
- *
- * mongoc_client_new --
- *
- *       Create a new mongoc_client_t using the URI provided.
- *
- *       @uri should be a MongoDB URI string such as "mongodb://localhost/"
- *       More information on the format can be found at
- *       http://docs.mongodb.org/manual/reference/connection-string/
- *
- * Returns:
- *       A newly allocated mongoc_client_t or NULL if @uri_string is
- *       invalid.
- *
- * Side effects:
- *       None.
- *
- *--------------------------------------------------------------------------
- */
 mongoc_client_t *
 mongoc_client_new (const char *uri_string)
 {
-   mongoc_topology_t *topology;
    mongoc_client_t *client;
-   mongoc_uri_t *uri;
+   bson_error_t error = {0};
+
+   if (!(client = mongoc_client_new_with_error (uri_string, &error)) &&
+       0 != strcmp (error.message, "")) {
```

Comment:
```suggestion
   if (!(client = mongoc_client_new_with_error (uri_string, &error))) {
```
`mongoc_client_new_with_error ()` should return `NULL` on error, implying `error.code != 0`, so the `strcmp()` seems unnecessary.